File: Problem-2/Python/solution_p2.py
def getFibSum(num):
    results = [] #initialize a list with "num" elements
    sum = 0
    fib = 0
    two_nums_ago = 1
    one_num_ago = 2
    while (fib < num):
        fib = one_num_ago + two_nums_ago
        if (fib > num):
            break
        if (fib % 2 == 0):
            sum += fib
        two_nums_ago = one_num_ago
        one_num_ago = fib
    if (sum != 0):
        sum += 2
    return sum

assert(getFibSum(10) == 10) # 1,2,3,5,8... 2 + 8 = 10
assert(getFibSum(100) == 44) #1,2,3,5,8,13,21,34,55,89... 2 + 8 + 34 = 44 

if __name__ == "__main__":
    print(getFibSum(4000000))

File: Problem-2/Java/solution_p2.java
public class solution_p2 {
    
    public static int fibSum(int number) {
        int twoNumsAgo = 1;
        int oneNumAgo = 2;
        int sum = 0;
        int fib = 3;
        while (fib < number) {
            fib = oneNumAgo + twoNumsAgo;
            if (fib > number) {
                break;
            }
            if (fib % 2 == 0) {
                sum += fib;
            }
            twoNumsAgo = oneNumAgo;
            oneNumAgo = fib;
        }
        if (sum > 0) {
            sum += 2;
        }
        return sum;
        
    }

    public static void main(String[] args) {
        assert fibSum(10) == 10 : "The sum of the even fibonacci numbers less than 10 should be equal to 10";
        assert fibSum(100) == 44 : "The sum of the even fibonacci numbers less than 100 should be equal to 44";
        System.out.println(fibSum(4000000)); // = 4613732
    }
}
